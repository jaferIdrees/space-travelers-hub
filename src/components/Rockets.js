import React from 'react';
import { useSelector } from 'react-redux';
// import Table from 'react-bootstrap/Table';
import { v4 as uuidv4 } from 'uuid';

function Rockets() {
  const rockets = useSelector((store) => store.rocketsReducer);
  return (
    rockets.map((rocket) => <p key={uuidv4()}>{rocket.rocket_name}</p>)
  );
}

export default Rockets;
