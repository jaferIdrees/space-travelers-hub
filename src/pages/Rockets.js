import React from 'react';
import RocketsList from '../components/Rockets';

const Rockets = () => (
  <RocketsList />
);

export default Rockets;
