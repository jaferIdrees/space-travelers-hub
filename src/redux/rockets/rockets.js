import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import GetRockets from '../../API/rocketsService';

const initialState = [];

export const retrieveRockets = createAsyncThunk(
  'rockets',
  async () => {
    const response = await GetRockets();
    const newState = [];
    response.data.forEach((rocket) => {
      const {
        // eslint-disable-next-line camelcase
        id, rocket_name, description, flickr_images,
      } = rocket;
      newState.push({
        // eslint-disable-next-line camelcase
        id, rocket_name, description, flickr_images,
      });
    });
    return newState;
  },
);

// Reducer
const rocketsSlice = createSlice({
  name: 'rockets',
  initialState,
  reducers: {
    reserveRocket: {
      reducer:
      (state, action) => state.map((rocket) => {
        if (rocket.id !== action.payload.rocket.id) {
          return rocket;
        }
        return { ...rocket, reserved: !rocket.reserved };
      }),
      prepare: (id) => ({
        payload: id,
      }),
    },
  },
  extraReducers: (builder) => {
    builder.addCase(retrieveRockets.fulfilled, (state, action) => action.payload);
  },
});

export const { reserveRocket } = rocketsSlice.actions;

export default rocketsSlice.reducer;
