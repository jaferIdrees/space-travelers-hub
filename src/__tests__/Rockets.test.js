import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
// import { render, screen, fireEvent } from '@testing-library/react';
import store from '../redux/configureStore';
import Rockets from '../components/Rockets';
import { retrieveRockets } from '../redux/rockets/rockets';

describe('Test rocket reducers and functions', () => {
  it('Fetches data from API', async () => {
    await store.dispatch(retrieveRockets());
    expect(store.getState().rockets).toBeTruthy();
  });
});

describe('Rockets component testing using snapshots', () => {
  it('Test Rockets component renders', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Rockets />
        </Provider>,
      );
    expect(tree).toMatchSnapshot();
  });
});
