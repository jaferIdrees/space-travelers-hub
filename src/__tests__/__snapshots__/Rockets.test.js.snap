// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets component testing using snapshots Test Rockets component renders 1`] = `
<link
  crossOrigin="anonymous"
  href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.0-beta1/dist/css/bootstrap.min.css"
  integrity="sha384-0evHe/X+R7YkIZDRvuzKMRqM+OrBnVFBL6DOitfPri4tjfHxaWutUpFmBp4vmVor"
  rel="stylesheet"
/>
`;
